benchmarks/entropy_adiabat/plugins/compute_entropy_profile.cc

```diff
@@ -0,0 +1,305 @@
+                  ExcInternalError());
+          return property.back();
+        }
+
+      if (z <= 0)
+        {
+          Assert (z >= -delta_z, ExcInternalError());
+          return property.front();
+        }
+
+      const double floating_index = z/delta_z;
+      const unsigned int i = static_cast<unsigned int>(floating_index);
+
+      // If p is close to an existing value use that one. This prevents
+      // asking for values at i+1 while initializing i+1 (when p is at the
+      // depth of index i).
+      if (std::abs(floating_index-std::floor(floating_index+0.5)) < 1e-6)
+        return property[i];
+
+      Assert (i+1 < property.size(), ExcInternalError());
+
+      // now do the linear interpolation
+      const double d = floating_index - i;
+      Assert ((d>=0) && (d<=1), ExcInternalError());
+
+      return d*property[i+1] + (1.-d)*property[i];
+    }
+
+
+
+
+    template <int dim>
+    void
+    ComputeEntropyProfile<dim>::declare_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection("Adiabatic conditions model");
+      {
+        prm.enter_subsection("Compute entropy profile");
+        {
+          prm.declare_entry ("Number of points", "1000",
+                             Patterns::Integer (5),
+                             "The number of points we use to compute the adiabatic "
+                             "profile. The higher the number of points, the more accurate "
+                             "the downward integration from the adiabatic surface "
+                             "conditions will be.");
+
+          prm.declare_entry ("Surface entropy", "0",
+                             Patterns::Double(),
+                             "The surface entropy for the profile.");
+        }
+        prm.leave_subsection();
+      }
+      prm.leave_subsection();
+    }
+
+
+    template <int dim>
+    void
+    ComputeEntropyProfile<dim>::parse_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection("Adiabatic conditions model");
+      {
+        prm.enter_subsection("Compute entropy profile");
+        {
+          n_points = prm.get_integer ("Number of points");
+          surface_entropy = prm.get_double ("Surface entropy");
+        }
+        prm.leave_subsection();
+      }
+      prm.leave_subsection();
+    }
+  }
+}
+
+
+// explicit instantiations
+namespace aspect
+{
+  namespace AdiabaticConditions
+  {
+    ASPECT_REGISTER_ADIABATIC_CONDITIONS_MODEL(ComputeEntropyProfile,
+                                               "compute entropy profile",
+                                               "A model in which the adiabatic profile is "
+                                               "calculated by solving the hydrostatic equations for "
+                                               "pressure and entropy in depth. "
+                                               "Of course the entropy along an adiabat is constant. "
```

Comment:
I modified the description of the benchmark. There is a transient gradient that moves through the model before it reaches steady-state (because the initial entropy is different from the entropy of the material that enters through the top boundary.